### ISSUE TITLE
Extra spacing between input-group-btns in a .modal-footer

### ISSUE THREAD

**Issue by maximepvrt** [twbs/bootstrap#18949]
at line 5973 :
```
.modal-footer .btn + .btn {
  margin-bottom: 0;
  margin-left: 5px;
}
```

![capture d ecran 2016-01-20 a 15 13 08](https://cloud.githubusercontent.com/assets/1840026/12451216/5d9d1a48-bf88-11e5-9005-72ff5cf7f10c.png)

**Comment by cvrebert** [comment]
@maximepvrt Please post a JS Bin or JS Fiddle that demonstrates the problem.

**Comment by maximepvrt** [comment]
https://jsfiddle.net/mfh63zjv/

**Comment by twbs-lmvtfy** [comment]
Hi @maximepvrt!

You appear to have posted a live example (https://fiddle.jshell.net/mfh63zjv/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 83, column 9: [E003](https://github.com/twbs/bootlint/wiki/E003): Found one or more `.row`s that were not children of a grid column or descendants of a `.container` or `.container-fluid`
* line 84, column 11: [E013](https://github.com/twbs/bootlint/wiki/E013): Only columns (`.col-*-*`) may be children of `.row`s
* line 85, column 13: [E014](https://github.com/twbs/bootlint/wiki/E014): Columns (`.col-*-*`) can only be children of `.row`s or `.form-group`s
* line 95, column 13: [E014](https://github.com/twbs/bootlint/wiki/E014): Columns (`.col-*-*`) can only be children of `.row`s or `.form-group`s

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by maximepvrt** [comment]
https://fiddle.jshell.net/mfh63zjv/2/

**Comment by twbs-lmvtfy** [comment]
Hi @maximepvrt!

You appear to have posted a live example (https://fiddle.jshell.net/mfh63zjv/2/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 85, column 11: [E013](https://github.com/twbs/bootlint/wiki/E013): Only columns (`.col-*-*`) may be children of `.row`s
* line 86, column 13: [E014](https://github.com/twbs/bootlint/wiki/E014): Columns (`.col-*-*`) can only be children of `.row`s or `.form-group`s
* line 96, column 13: [E014](https://github.com/twbs/bootlint/wiki/E014): Columns (`.col-*-*`) can only be children of `.row`s or `.form-group`s

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by maximepvrt** [comment]
https://fiddle.jshell.net/maximepvrt/7pmL9tc6/1/

**Comment by mdo** [comment]
Yup, looks like the `.modal-footer .btn + .btn` selector is interfering here and adding that `5px` on the left.

**Comment by mdo** [comment]
Merged #19137.

**Comment by cvrebert** [comment]
@mdo What about v3?

**Comment by maximepvrt** [comment]
This bug is in v3 too

**Comment by Lausselloic** [comment]
Hello,
The fix applied onto the V4-dev is quite hard "remove margin for every buttons in modal footers". Why not, it's a major breaking version. But for V3, to ensure compatibility, I propose to just also override the margin-left for .input-group-btn into modal.less line 115 : 
// but override that for button groups
  .input-group-btn,
  .btn-group {
    .btn + .btn {
      margin-left: -1px;
    }
  }
Instead of : 
// but override that for button groups
  .btn-group  .btn + .btn {
    margin-left: -1px;
  }

Will fix this issue, and ensure compatibility with previous versions.

Regards

**Comment by mdo** [comment]
[**Bootstrap 3 is no longer being officially developed or supported.**](https://github.com/twbs/bootstrap/issues/20631)

All work has moved onto our next major release, v4. As such, this issue or pull request is being closed as a "won't fix." For additional help and support, we recommend [utilizing our community resources](https://github.com/twbs/bootstrap/blob/master/README.md#community). Thanks for your understanding, and see you on the other side of v4!

<3,
@mdo and team
